packages/beacon-node/src/chain/chain.ts

```diff
@@ -463,28 +463,52 @@ export class BeaconChain implements IBeaconChain {
         return {block: data, executionOptimistic: isOptimisticBlock(block)};
       }
       // If block is not found in hot db, try cold db since there could be an archive cycle happening
-      // TODO: Add a lock to the archiver to have determinstic behaviour on where are blocks
+      // TODO: Add a lock to the archiver to have deterministic behavior on where are blocks
     }
 
     const data = await this.db.blockArchive.getByRoot(fromHexString(root));
     return data && {block: data, executionOptimistic: false};
   }
 
+  async produceBlockBodyPhase0(
+    blockAttributes: Pick<BlockAttributes, "slot" | "graffiti" | "randaoReveal">
+  ): Promise<phase0.BeaconBlockBody> {
+    const {slot} = blockAttributes;
+    const head = this.forkChoice.getHead();
+    const state = await this.regen.getBlockSlotState(
+      head.blockRoot,
+      slot,
+      {dontTransferCache: true},
+      RegenCaller.produceBlock
+    );
+
+    // TODO: To avoid breaking changes for metric define this attribute
+    const blockType = BlockType.Full;
```

Comment:
To avoid changes in the metrics structure for now in this PR I had to fix this type for now. We may think to refactor the metrics structure after merging this PR.

Reply:
Shouldn't metric consolidation be part of this PR, feel like passing blockType around just for metrics is not ideal